File: PeaceDatabase/Core/Models/Document.cs
using System.Collections.Generic;

namespace PeaceDatabase.Core.Models
{

    public class Document
    {
        public string Id { get; set; } = string.Empty; // "_id"
        public string? Rev { get; set; }               // "_rev"
        public bool Deleted { get; set; }              // "_deleted"
        // Поля для равенств/диапазонов:
        public Dictionary<string, object>? Data { get; set; }
        // Мета-теги:
        public List<string>? Tags { get; set; }

        // Полнотекстовый контент:
        public string? Content { get; set; }
    }
}

File: spark/peacedb-datasource/src/main/java/io/peacedb/spark/model/Document.java
package io.peacedb.spark.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Document {
    @JsonProperty("id")
    public String id;
    @JsonProperty("rev")
    public String rev;
    @JsonProperty("deleted")
    public boolean deleted;
    @JsonProperty("data")
    public Map<String, Object> data;
    @JsonProperty("tags")
    public List<String> tags;
    @JsonProperty("content")
    public String content;
}
